packages/@uppy/store-default/src/index.ts

```diff
@@ -0,0 +1,49 @@
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore We don't want TS to generate types for the package.json
+import packageJson from '../package.json'
+
+export type GenericState = Record<string, unknown>
+
+export type Listener<T> = (
+  prevState: T,
+  nextState: T,
+  patch: Partial<T>,
+) => void
+
+/**
+ * Default store that keeps state in a simple object.
+ */
+class DefaultStore<T extends GenericState = GenericState> {
+  static VERSION = packageJson.version
+
+  public state: T = {} as T
```

Comment:
now that I see this type it's actually odd we don't have an initial state passed in the constructor, which would make this type and state correct from the start. Just a thought, not something for now.